addons/payment_asiapay/models/payment_transaction.py

```diff
@@ -0,0 +1,139 @@
+from odoo.tools.float_utils import float_compare
+
+from .const import CURRENCY_MAPPING
+from odoo.addons.payment_asiapay import utils as payment_asiapay_utils
+from odoo.addons.payment_asiapay.controllers.main import AsiaPayController
+
+_logger = logging.getLogger(__name__)
+
+class PaymentTransaction(models.Model):
+    _inherit = "payment.transaction"
+
+    def _get_specific_rendering_values(self, processing_values):
+        """Override of payment to return asiapay rendering values.
+
+        Note: self.ensure_one() from `_get_processing_values`
+
+        :param dict processing_values: The generic and specific processing values of the transaction
+        :return: The dict of acquirer-specific processing values
+        :rtype: dict
+        """
+        res = super()._get_specific_rendering_values(processing_values)
+        if self.provider != "asiapay":
+            return res
+
+        base_url = self.acquirer_id.get_base_url()
+        order_ref = "{} {}".format(self.reference, datetime.now().strftime("%m/%d/%Y %H:%M:%S"))
+        rendering_values = {
+            "merchantId": self.acquirer_id.asiapay_merchant_id,
+            "amount": self.amount,
+            "orderRef": order_ref,
+            "currCode": CURRENCY_MAPPING.get(self.acquirer_id.asiapay_currency_id.name),
+            "mpsMode": "SCP",
+            "successUrl": urls.url_join(base_url, AsiaPayController._return_url),
+            "failUrl": urls.url_join(base_url, AsiaPayController._return_url),
+            "cancelUrl": urls.url_join(base_url, AsiaPayController._return_url),
+            "payType": "N",
+            "lang": payment_asiapay_utils.get_lang(self.env.lang),
+            "payMethod": "ALL",
+        }
+        if self.acquirer_id.asiapay_currency_id.id != self.currency_id.id:
+            rendering_values.update({
+                "foreignCurrCode": CURRENCY_MAPPING.get(self.currency_id.name),
+            })
+        if self.acquirer_id.asiapay_secure_hash:
+            args = {
+                'hash_function': self.acquirer_id.asiapay_secure_hash_function,
+                'merchant_id': self.acquirer_id.asiapay_merchant_id,
+                'merchant_reference': order_ref,
+                'curr_code': CURRENCY_MAPPING.get(self.acquirer_id.asiapay_currency_id.name),
+                "amount": str(self.amount),
+                'payment_type': "N",
+                'secret': self.acquirer_id.asiapay_secure_hash,
+            }
+            secure_hash = payment_asiapay_utils.generate_secure_hash(**args)
+            rendering_values.update({
+                "secureHash": secure_hash,
+            })
+
+        rendering_values.update({
+            "api_url": self.acquirer_id._asiapay_get_api_url()
+        })
+        return rendering_values
+
+    @api.model
+    def _get_tx_from_feedback_data(self, provider, data):
+        tx = super()._get_tx_from_feedback_data(provider, data)
+        if provider != "asiapay":
```

Comment:
This prevents searching for the transaction again if the method is already called on the transaction record like it can typically be done in controllers.
```suggestion
        if provider != "asiapay" or len(tx) == 1:
```